### DISCUSSION TITLE
Connection encryption

### DISCUSSION THREAD

**Discussion by allaert** [modernuo/ModernUO#1407] [Q&A] [answered]
Hi,

I've been looking into running a UO server and have been trying out ModernUO.
I was wondering if there is any way to turn on connection encryption cause what i noticed on my shard is that the login password is just sent unencrypted over the network.

Thanks for any info you can share.

**Comment by kamronbatman** [comment, marked as answer]
We don't have UO encryption support yet.
There is an option in the code to add an Encoder and Decoder. The encryption for UO is not that great, it's twofish for new clients and blowfish for older clients using a known shared key.
